src/hal/i_components/time.icomp

```diff
@@ -83,15 +83,15 @@ function _;
 
 FUNCTION(_)
 {
-    __u32 totalseconds;
-    if(start && !old_start) totalnsec = 0;
+    hal_u32_t totalseconds;
+    if(start && !old_start) totalnsec = 0.0;
 
     if(start){
     totalnsec = totalnsec + period;
     totalseconds = totalnsec * 0.000000001;
```

Comment:
Implicit typecast to floating point value is happening here Should be replaced by integer division.
`totalseconds = (hal_u32_t)(totalnsec / 1e9)`